pkg/machines/scripts/create_machine.sh

```diff
@@ -41,10 +26,17 @@ else
     DISK_OPTIONS="size=$STORAGE_SIZE,format=qcow2"
 fi
 
-GRAPHICS_PARAM="--graphics vnc,listen=127.0.0.1"
-if spiceSupported; then
+DOM_GRAPHICS_CAPABILITIES="$(virsh domcapabilities | awk "/<graphics supported='yes'/{flag=1; next}/<\/graphics/{flag=0}flag")"
+GRAPHICS_PARAM=""
+if echo "$DOM_GRAPHICS_CAPABILITIES" | grep -q spice; then
+    GRAPHICS_PARAM="--graphics vnc,listen=127.0.0.1"
+fi
+if echo "$DOM_GRAPHICS_CAPABILITIES" | grep -q vnc; then
     GRAPHICS_PARAM="--graphics spice,listen=127.0.0.1 $GRAPHICS_PARAM"
```

Comment:
This is the wrong way around; you first grep for spice and then configure vnc, then vice versa.